notify-send.sh

```diff
@@ -200,109 +200,172 @@ process_special_action() {
     action_key="$1"
     command="$2"
 
-    if [[ -z "$action_key" ]] || [[ -z "$command" ]]; then
+    if [ -z "$action_key" ] || [ -z "$command" ]; then
         echo "Command must not be empty"
         exit 1
     fi
 
     ACTION_COMMANDS=("${ACTION_COMMANDS[@]}" "$action_key" "$command")
 
-    if [[ "$action_key" != close ]]; then
+    if [ "$action_key" != close ]; then
         local action="$(make_action "$action_key" "$name")"
         ACTIONS=("${ACTIONS[@]}" "$action")
     fi
 }
 
 process_posargs() {
-    if [[ "$1" = -* ]] && ! [[ "$positional" = yes ]] ; then
-        echo "Unknown option $1"
-        exit 1
+    case "$1" in
+        -*) if [ "$positional" != yes ] ; then
+                echo "Unknown option $1"
+                exit 1
+            fi ;;
+    esac
+
+    if [ "$SUMMARY_SET" = n ]; then
+        SUMMARY="$1"
+        SUMMARY_SET=y
     else
-        if [[ "$SUMMARY_SET" = n ]]; then
-            SUMMARY="$1"
-            SUMMARY_SET=y
-        else
-            BODY="$1"
-        fi
+        BODY="$1"
     fi
 }
 
-while (( $# > 0 )) ; do
+while [ $# -gt 0 ] ; do
```

Comment:
This kind of bracket expansion does not exist in POSIX shell.  The replacement is `test`'s numeric comparison operators.  In this case, `-gt`, for greater-than.  An obviously symmetric `-lt` exists, as well as x-than-or-equal variants too.